Added UnitTests to the project

StraussLukas opened the pull request.

StraussLukas committed:
8 UnitTests were written. 8/8 are successful.

StraussLukas committed:
12 UnitTests were written. 11/12 are successful.

StraussLukas committed:
16 UnitTests were written. 14/16 are successful.

StraussLukas committed:
README updated.

StraussLukas commented on src/test/java/stage1/RunnerSprintNameGeneratorTest.java:
```diff
@@ -0,0 +1,291 @@
+	
+	}
+	@Test
+	public void test_markWinnerName_checkIfCorrectNameIsHighlightedGreen() {
```
UnitTest not successful

GeroHerkenrath replied:
Well, that's to be expected from the logic you wrote. :)
So in line 253 in this test you call `RunnerSprintNameGenerator.getNamesWithMostVotes()`. If you look up that method, it executes `clearHighlighting()` on each `SprintName` object the runner has in its static member array, so the background color that was set in `markWinnerName(RunnerSprintNameGenerator.getNamesWithMostVotes())` gets reset to white again.

In the broader context that's the result of mixing something like a getter (more or less) and UI update code.
For now you can resolve this by not getting the sprint name you are interested in via `getNamesWithMostVotes().get(0)` (in the assert), but instead via `getSprintNames().get(1)`. That does not change the color and will make the test pass

StraussLukas commented on src/test/java/stage1/RunnerSprintNameGeneratorTest.java:
```diff
@@ -0,0 +1,291 @@
+	}
+	
+	@Test
+	public void test_markNamesDraw_checkIfNamesWereHighlightedYellow() {
```
UnitTest not successful

GeroHerkenrath replied:
I haven't checked due to time yet, but I am pretty sure that this is due to the same issue as in test `test_markWinnerName_checkIfCorrectNameIsHighlightedGreen`, so it should be resolvable via the same means.

GeroHerkenrath reviewed (changes requested):
Once both tests are repaired finish the draft and I will review again (maybe also in peson with you).

StraussLukas committed:
JUnit dependency added.

StraussLukas committed:
Testfailures resolved: 16/16 Unittest are successful.
SetUp-method and tearDown-method added in RunnerSprintNameGeneratorTest.
Typing mistakes resolved.
